.github/shared/src/changed-files.js

```diff
@@ -44,6 +44,125 @@ export async function getChangedFiles(options = {}) {
   return files;
 }
 
+/**
+ * @param {Object} [options]
+ * @param {string} [options.baseCommitish] Default: "HEAD^".
+ * @param {string} [options.cwd] Current working directory.  Default: process.cwd().
+ * @param {string} [options.headCommitish] Default: "HEAD".
+ * @param {import('./logger.js').ILogger} [options.logger]
+ * @returns {Promise<{additions: string[], modifications: string[], deletions: string[], renames: {from: string, to: string}[], total: number}>}
+ */
+export async function getChangedFilesStatuses(options = {}) {
```

Comment:
Do you really needed the statuses?  I was hoping we could simplify by just using a flat list of changed files.

If you do need statuses, instead of adding another, we should probably update the existing API to return statuses.

I will refactor this in a later PR.

Reply:
@mikeharder 
This is [the place](https://github.com/Azure/azure-rest-api-specs/pull/35353/files#diff-1ee83d4d127867a6377ccbcc8cdc38f4a9d8a938e4d958f1c20938a22613bbedR52) in breaking change tool to get changed files in statuses. Based on different statues, the tool needs to create dummy file for comparison.

As a shared library, it would be better to return them in different lists instead of requiring each caller to parse the result.

We still have other usages which don't care about the status. I'd rather keep both APIs but extending the first API to take an option of status filter.